src/xinput.cpp

```diff
@@ -0,0 +1,61 @@
+/***************************************************************************
+* Copyright (c) 2018, Martin Renou, Johan Mabille, Sylvain Corlay and      *
+* Wolf Vollprecht                                                          *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include <iostream>
+#include <stdexcept>
+#include <string>
+
+#include "xinput.hpp"
+
+#include "xeus/xinterpreter.hpp"
+
+namespace xcpp
+{
+    std::string input_request(const std::string& prompt, bool password)
+    {
+        auto& interpreter = xeus::get_interpreter();
+
+        // Register the input handler
+        std::string value;
+        interpreter.register_input_handler([&value](const std::string& v) {
+            value = v;
+        });
+
+        // Send the input request
+        interpreter.input_request(prompt, password);
+
+        // Remove input handler
+        interpreter.register_input_handler(nullptr);
+
+        return value;
+    }
```

Comment:
This function is the same as [in xeus-python](https://github.com/QuantStack/xeus-python/blob/3f5899c280b8f03631cd27f02cba771ad3c27095/src/xinput.cpp#L26-L47) and we intend on moving it to xeus core.

The code is simpler than [the one of ipykernel](https://github.com/ipython/ipykernel/blob/118b684f75d51d72819ca90b69a245e9b02067a5/ipykernel/kernelbase.py#L856-L894), but we don't have specific handling of EOF.

Reply:
cc @minrk do you know in which case the EOF case handled by this code could happen?

https://github.com/ipython/ipykernel/blob/118b684f75d51d72819ca90b69a245e9b02067a5/ipykernel/kernelbase.py#L891-L894

Reply:
\x04 is the ASCII end-of-text byte, used to propagate EOF from a frontend to the kernel. e.g. piping input through a program sending input to the kernel. This may literally never happen in practice (it can't happen in the notebook, but might possibly in jupyter-console).

Reply:
OK, thanks for chiming in!

`\x04` is a valid character for an `std::string`.  It is actually a case where strlen and string.size may give different results.